main.go

```diff
@@ -169,3 +173,20 @@ func main() {
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+// newClientFunc returns a client reads from cache and write directly to the server
+// this avoid get unstructured object directly from the server
+// see issue: https://github.com/kubernetes-sigs/cluster-api/issues/1663
+func newClientFunc(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
```

Comment:
Would it make sense to expose this function as part of the util package so providers might use it as well?

Reply:
i think it should be a exposed function by controller-runtime, not cluster-api